src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -123,18 +123,37 @@ public class FailoverConnectionPlugin implements IConnectionPlugin {
   private final PropertySet propertySet;
   private final IConnectionPlugin nextPlugin;
   private final Log logger;
+
+  // patterns
   private final Pattern auroraDnsPattern =
```

Comment:
Looks like these Regex patterns and methods are already implemented in RdsUtils, but FailoverConnectionPlugin is calling the outdated methods in this file instead. Should we centralize these methods?